homeassistant/components/zestimate/sensor.py

```diff
@@ -115,12 +115,16 @@ def update(self):
             return
         data = data_dict['response'][NAME]
         details = {}
-        details[ATTR_AMOUNT] = data['amount']['#text']
-        details[ATTR_CURRENCY] = data['amount']['@currency']
-        details[ATTR_LAST_UPDATED] = data['last-updated']
-        details[ATTR_CHANGE] = int(data['valueChange']['#text'])
-        details[ATTR_VAL_HI] = int(data['valuationRange']['high']['#text'])
-        details[ATTR_VAL_LOW] = int(data['valuationRange']['low']['#text'])
+        if data['amount'] is not None:
```

Comment:
So if the key does not exist, this will raise. Is each key always set to `None` if it's not there?

Reply:
That would be up to the vendor (Zillow in this case) and their documentation doesn't say how they handle it.  Setting valueChange field to none is either an API change on their side or a bug on their side.  I guess it's a question of how much validation is reasonable to have on the API consumer side.  I didn't want to just remove their field to get it working again just in case they start passing the value again in the future and while I was tweaking the code I figured I would check the other fields being pulled to.

Reply:
Oh, and your first question?  No, it doesnt catch the case of the key missing.  Let me add that check and update.

Reply:
@#$%#$%  the last commit I made to this pull request had a bunch of other commits.  Not sure there's a way for me to fix without closing this pull request and opening a new one?